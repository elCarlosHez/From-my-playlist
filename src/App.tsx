import React from 'react';

import { Link } from 'react-router-dom';

import logo from './assets/logo.png';
import './styles/App.css';

const App = () => (
  <div className="flex justify-center items-center flex-col">
    <img className="max-w-full mb-5" alt="From my music" src={logo} />
    <div className="flex justify-between items-center">
      <Link to="/youtube" className="btn btn-primary text-center flex-1 ml-5">
        Link Youtube Music Account
      </Link>

      <Link to="/spotify" className="btn btn-secondary text-center flex-1 ml-5">
        Link Spotify Account
      </Link>
    </div>
  </div>
);

export default App;
