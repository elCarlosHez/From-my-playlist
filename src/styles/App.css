@import "./components/theme.css";
@import "./components/buttons.css";
@tailwind base;
@tailwind components;
@tailwind utilities;
