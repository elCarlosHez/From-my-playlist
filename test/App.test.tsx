import React from 'react';
import renderer from 'react-test-renderer';
import App from '../src/App';

function toJson(component: renderer.ReactTestRenderer) {
  const result = component.toJSON()
  expect(result).toBeDefined()
  expect(result).not.toBeInstanceOf(Array)
  return result as renderer.ReactTestRendererJSON
}

describe("App tests", () => {
  it('Renders without crash', () => {
    const component = renderer.create(
      <App />,
    )
    let tree = toJson(component)
    console.log(tree)
    expect(tree).toMatchSnapshot()
  })
})
