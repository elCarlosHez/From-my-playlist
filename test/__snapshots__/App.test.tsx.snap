// Vitest Snapshot v1

exports[`App tests > Renders without crash 1`] = `
<div
  className="flex justify-center items-center flex-col"
>
  <img
    className="max-w-full mb-5"
    src="/src/assets/logo.png"
  />
  <div
    className="flex justify-between"
  >
    <button
      className="btn btn-primary flex-1"
    >
      Link Youtube Music Account
    </button>
    <button
      className="btn btn-secondary flex-1 ml-5"
    >
      Link Spotify Account
    </button>
  </div>
</div>
`;

exports[`Renders without crash 1`] = `
<div
  className="flex justify-center items-center flex-col"
>
  <img
    className="max-w-full mb-5"
    src="/src/assets/logo.png"
  />
  <div
    className="flex justify-between"
  >
    <button
      className="btn btn-primary flex-1"
    >
      Link Youtube Music Account
    </button>
    <button
      className="btn btn-secondary flex-1 ml-5"
    >
      Link Spotify Account
    </button>
  </div>
</div>
`;
